includes/ucf-social-common.php

```diff
@@ -102,19 +116,164 @@ public static function has_social_feed( $content ) {
 			// [ucf-social-feed] shortcode, as well as processed string
 			// contents for substrings that are likely to be present in social
 			// feed templates
-			if (
-				has_shortcode( $content, 'ucf-social-feed' )
-				|| ( strpos( $content_processed, 'new Curator.' ) !== false || strpos( $content_processed, 'ucf-social-feed' ) !== false )
-			) {
+			if ( has_shortcode( $content, 'ucf-social-feed' ) || strpos( $content_processed, 'ucf-social-feed' ) !== false ) {
 				$has_feed = true;
 			}
 
 			return $has_feed;
 		}
 
+		/**
+		 * Returns a JSON object from the provided URL.  Detects undesirable status
+		 * codes and returns false if the response doesn't look valid.
+		 *
+		 * @since 3.0.0
+		 * @author Jo Dickson
+		 * @param string $url URL that points to a JSON object/feed
+		 * @return mixed JSON-decoded object or false on failure
+		 */
+		private static function fetch_json( $url ) {
+			$response      = wp_remote_get( $url, array( 'timeout' => 15 ) );
+			$response_code = wp_remote_retrieve_response_code( $response );
+			$result        = false;
+
+			if ( is_array( $response ) && is_int( $response_code ) && $response_code < 400 ) {
+				$result = json_decode( wp_remote_retrieve_body( $response ) );
+			}
+
+			return $result;
+		}
+
+		/**
+		 * Retrieves data from Curator.io's API based on the API key provided
+		 * in plugin options.
+		 * Caches and references transient data.
+		 *
+		 * @author Jo Dickson
+		 * @since 3.0.0
+		 * @return array
+		 */
+		private static function get_social_feed_data() {
+			$transient_name = 'ucf_social_curator_api_data';
+			$transient      = get_transient( $transient_name );
+			$result         = array();
+
+			if ( $transient === false ) {
+				$api_key = get_option( UCF_Social_Config::$option_prefix . 'curator_api_key' );
+
+				// Back out if we don't have an API key to use
+				if ( empty( $api_key ) ) {
+					return $result;
+				}
+
+				$url = 'https://api.curator.io/v1/feeds?api_key=' . $api_key;
+				$response = self::fetch_json( $url );
+
+				if ( $response ) {
+					$result = $response;
+					set_transient( $transient_name, $result, 12 * HOUR_IN_SECONDS );
+				}
+			}
+			else if ( is_array( $transient ) ) {
+				$result = $transient;
+			}
+
+			return $result;
+		}
+
+		/**
+		 * Retrieves options set on the given feed within Curator.io.
+		 *
+		 * @author Jo Dickson
+		 * @since 3.0.0
+		 * @param string $feed_id | ID for a feed from curator.io
+		 * @return object
+		 */
+		private static function get_social_feed_option_data( $feed_id ) {
+			$data = self::get_social_feed_data();
+			$options = '{}';
+
+			foreach ( $data as $feed_obj ) {
+				if ( strtolower( $feed_obj->public_key ) === strtolower( $feed_id ) ) {
+					$options = $feed_obj->widget_options;
+					break;
+				}
+			}
+
+			return json_decode( $options );
```

Comment:
We should potentially throw this in a try/catch to prevent an exception from being thrown if there's something funky in the json.